string set_node
---
string next_node
